src/sentry/incidents/logic.py

```diff
@@ -812,17 +820,15 @@ def update_alert_rule(
             comparison_delta = int(timedelta(minutes=comparison_delta).total_seconds())
 
         updated_fields["comparison_delta"] = comparison_delta
-
-    # if we modified the comparison_delta or the time_window, we should update the resolution accordingly
-    if "comparison_delta" in updated_fields or "time_window" in updated_query_fields:
-        if comparison_delta is not None:
-            detection_type = AlertRuleDetectionType.PERCENT
-        else:
-            if seasonality:
-                detection_type = AlertRuleDetectionType.DYNAMIC
+    if detection_type is None:
+        if "comparison_delta" in updated_fields:  # some value changed -> update type if necessary
```

Comment:
Can we combine this into one line?

Reply:
I tried, and it causes tests to fail, so apparently not